great_tables/_formats.py

```diff
@@ -663,112 +778,175 @@ def fmt_scientific(
 
-        if exp_style == "x10n":
-            # Define the exponent string based on the `exp_style` that is the default
-            # ('x10n'); this is styled as 'x 10^n' instead of using a fixed symbol like 'E'
+    # Remove trailing zeros and decimal marks from the `m_part`
+    if drop_trailing_zeros:
+        m_part = m_part.rstrip("0")
+    if drop_trailing_dec_mark:
+        m_part = m_part.rstrip(".")
 
-            # Determine which values don't require the (x 10^n) for scientific formatting
-            # since their order would be zero
-            small_pos = _has_sci_order_zero(value=x)
+    # Force the positive sign to be present if the `force_sign_m` option is taken
+    if is_positive and force_sign_m:
+        m_part = "+" + m_part
 
-            # Force the positive sign to be present if the `force_sign_n` option is taken
-            if force_sign_n and not _str_detect(n_part, "-"):
-                n_part = "+" + n_part
+    if exp_style == "x10n":
+        # Define the exponent string based on the `exp_style` that is the default
+        # ('x10n'); this is styled as 'x 10^n' instead of using a fixed symbol like 'E'
 
-            # Implement minus sign replacement for `m_part` and `n_part`
-            m_part = _replace_minus(m_part, minus_mark=minus_mark)
-            n_part = _replace_minus(n_part, minus_mark=minus_mark)
+        # Determine which values don't require the (x 10^n) for scientific formatting
+        # since their order would be zero
+        small_pos = _has_sci_order_zero(value=x)
 
-            if small_pos:
-                # If the value is small enough to not require the (x 10^n) notation, then
-                # the formatted value is based on only the `m_part`
-                x_formatted = m_part
-            else:
-                # Get the set of exponent marks, which are used to decorate the `n_part`
-                exp_marks = _context_exp_marks()
+        # Force the positive sign to be present if the `force_sign_n` option is taken
+        if force_sign_n and not _str_detect(n_part, "-"):
+            n_part = "+" + n_part
 
-                # Create the formatted string based on `exp_marks` and the two `sci_parts`
-                x_formatted = m_part + exp_marks[0] + n_part + exp_marks[1]
+        # Implement minus sign replacement for `m_part` and `n_part`
+        m_part = _replace_minus(m_part, minus_mark=minus_mark)
+        n_part = _replace_minus(n_part, minus_mark=minus_mark)
 
+        if small_pos:
+            # If the value is small enough to not require the (x 10^n) notation, then
+            # the formatted value is based on only the `m_part`
+            x_formatted = m_part
         else:
-            # Define the exponent string based on the `exp_style` that's not the default
-            # value of 'x10n'
+            # Get the set of exponent marks, which are used to decorate the `n_part`
+            exp_marks = _context_exp_marks(context=context)
```

Comment:
The `_context_exp_marks()` function is new here since scientific notation formatting is different in HTML and LaTeX. This follows the convention of getting the correct text fragment(s) for the rendering context.